scripts/full_check.sh

```diff
@@ -0,0 +1,111 @@
+#! /bin/bash
+
+# Copyright (C) 2024 Intel Corporation
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+# http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+# 
+# SPDX-License-Identifier: Apache-2.0
+
+
+# this is for developer local check only.
+# not used by github action
+
+check_tool() {
+  which $1 &> /dev/null || (echo "$1 not found!" && exit 1)
+}
+
+for arg in "$@"; do
+  case $arg in
+    -f|--format)
+      CHECK_FORMAT=1
+      ;;
+    -t|--tidy)
+      CHECK_TIDY=1
+      ;;
+    -l|--license)
+      CHECK_LICENSE=1
+      ;;
+    -c|--clean)
+      CLEANUP=1
+      ;;
+    *) # Handle other unknown options
+      echo "Unknown option: $1"
+      exit 1
+      ;;
+  esac
+done
+
+# set PR_REF if your pr target is not main
+: ${PR_REF:=main}
+
+PROJECT_ROOT="$(cd "$(dirname "$0")" && pwd)/../"
+cd "$PROJECT_ROOT"
+
+MERGE_BASE=$(git merge-base $PR_REF HEAD)
+CHANGED_FILES=$(git diff --name-only $MERGE_BASE HEAD)
+
+# if you do not have clang/clang++/clang-tidy/clang-format
+# please install it
+# conda install -c conda-forge clangxx cxx-compiler clang-tools
+
+if [ -n "$CHECK_TIDY" ]; then
+  echo "start tidy check..."
+  if [ -z "${MLIR_DIR}" ]; then
+    echo "The environment variable MLIR_DIR is not set."
+    exit 1
+  fi
+  check_tool "clang"
+  check_tool "clang++"
+  check_tool "clang-tidy"
+  check_tool "lit"
+
+  TIDY_ROOT="${PROJECT_ROOT}/build/tidy"
+  [ -n "$CLEANUP" ] && rm -rf "${TIDY_ROOT}"
+  mkdir -p "${TIDY_ROOT}"
+  cd "${TIDY_ROOT}"
+
+  cmake ../../ \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DMLIR_DIR=${MLIR_DIR} \
+    -DCMAKE_EXPORT_COMPILE_COMMANDS=True \
+    -DCMAKE_C_COMPILER=$(which clang) \
+    -DCMAKE_CXX_COMPILER=$(which clang++) \
+    -DLLVM_EXTERNAL_LIT=$(which lit)
+
+  for f in $(find ./include -name Makefile); do
+    targets=$(make -f $f help |grep IncGen); 
+    if [[ $? -eq 0 ]]; then 
```

Comment:
```suggestion
    if [ $? = 0 ]; then 
```

Reply:
bash `if` command checks the command exit code by default, the code above is like the check `if mybool == true`

please use either `if cmd` or `cmd && some_command`